ts-client/src/vault/index.ts

```diff
@@ -1,82 +1,67 @@
-        const allVaultsPromises = allVaultsInfo.map(async (vault) => {
-            const tokenInfo = tokenResolver.find(token => token.address === vault.token_address);
-            const apyStateResponse = await fetch(`${this.URL}/apy_state/${vault.token_address}`);
-            const apyState = await apyStateResponse.json() as VaultAPY;
+    const allVaultsInfoResponse = await fetch(`${this.URL}/vault_info`);
+    const allVaultsInfo = (await allVaultsInfoResponse.json()) as VaultInfo[];
+    const allVaultsPromises = allVaultsInfo.map(async (vault) => {
+      // const apyStateResponse = await fetch(`${this.URL}/apy_state/${vault.token_address}`);
+      // const apyState = await apyStateResponse.json() as VaultAPY;
 
-            if (tokenInfo && apyState) {
-                return {
-                    data: {
-                        ...vault,
-                        apyState,
-                    },
-                    vault: await VaultImpl.create(
-                        program,
-                        { baseTokenMint: new PublicKey(tokenInfo.address), baseTokenDecimals: tokenInfo.decimals }
-                    )
-                };
-            } else {
-                console.log({ error: `TokenInfo for ${vault.token_address} not found!` })
-                return null;
-            };
-        })
-        return allVaultsPromises;
-    }
+      return await VaultImpl.create(program, vault);
+    });
+    return allVaultsPromises;
+  }
 
-    public static async create(connection: Connection) {
-        const allVaultsPromises = await this.fetchAllVaultInfo(connection);
-        const allVaults = (await Promise.all(allVaultsPromises)).filter(Boolean) as AllVaults;
-        return new Vaults(connection, allVaults);
-    }
+  public static async create(connection: Connection) {
+    const allVaultsPromises = await this.fetchAllVaultInfo(connection);
+    const allVaults = (await Promise.all(allVaultsPromises)).filter(
+      Boolean
+    ) as AllVaults;
+    return new Vaults(connection, allVaults);
+  }
 
-    // Refetch all vault_info data, and merge them.
-    async refetch() {
-        const allVaultsPromises = await Vaults.fetchAllVaultInfo(this.connection);
-        const allVaults = (await Promise.all(allVaultsPromises)).filter(Boolean) as AllVaults;
+  // Refetch all vault_info data, and merge them.
+  async refetch() {
+    const allVaultsPromises = await Vaults.fetchAllVaultInfo(this.connection);
+    const allVaults = (await Promise.all(allVaultsPromises)).filter(
+      Boolean
+    ) as AllVaults;
 
-        allVaults.forEach(vault => {
-            const found = this.allVaults.findIndex(item => item.data.token_address === vault.data.token_address);
-            if (found > -1) {
-                this.allVaults[found].data = vault.data;
-            }
-        })
+    this.allVaults = [...allVaults];
 
-        return this.allVaults;
-    }
+    return this.allVaults;
+  }
```

Comment:
the reason the previous commit were doing 
```
this.allVaults.findIndex(item => item.data.token_address === vault.data.token_address);
             if (found > -1) {
                 this.allVaults[found].data = vault.data;
             }
```

was to prevent creating `VaultImpl` over and over again